libraries/checkers/obs_checker.py

```diff
@@ -17,35 +18,45 @@ def run(self):
         self.converted_dir is the directory of converted files
         :return:
         """
-        for chapter in range(1, 51):
-            filename = os.path.join(self.converted_dir, str(chapter).zfill(2)+'.html')
+        for chapter in range( 1, 51 ):
+            chapter_number = str(chapter).zfill(2)                                       # chapter check
+            filename = os.path.join(self.preconvert_dir, chapter_number + '.md')
 
             if not os.path.isfile(filename):
-                self.log.warning('Chapter {0} does not exist!'.format(chapter))
+                self.log.warning('Chapter {0} does not exist!'.format(chapter_number))
                 continue
 
-            chapter_html = read_file(filename)
-            soup = BeautifulSoup(chapter_html, 'html.parser')
+            chapter_md = read_file(filename)
+            is_title = chapter_md.find('# ')
+
+            if is_title < 0:                                                             # Find chapter headings
+                self.log.warning('Chapter {0} does not have a title!'.format(chapter_number))
+
+            expected_frame_count = obs_data['chapters'][str(chapter).zfill(2)]['frames'] # Identify missing frames
+
+            for frame_idx in range( 1, expected_frame_count ):
+                frame_index = str(frame_idx).zfill(2)
+                pattern = '-{0}-{1}.'.format(chapter_number, frame_index)
 
-            if not soup.find('body'):
-                self.log.warning('Chapter {0} has no body!'.format(chapter))
-                return
+                if chapter_md.find( pattern ) < 0:
+                    self.log.warning('Missing frame: {0}-{1}'.format(chapter_number, frame_index))
 
-            content = soup.body.find(id='content')
 
-            if not content:
-                self.log.warning('Chapter {0} has no content!'.format(chapter))
-                return
+            if re.search( r'^_.*:.*_[ ]*$', chapter_md, re.M ) == None:                                     # look for verse reference
```

Comment:
That is a helpful optimization from Python!  It still makes me nervous, especially in the Lambda environment that we don't have a lot of visibility into.

You are definitely correct that there is likely to be no noticeable speed improvement here, as we check an OBS project.